lib/Controller/SignallingController.php

```diff
@@ -259,4 +251,149 @@ protected function getUsersInRoom(Room $room) {
 
 		return $usersInRoom;
 	}
+
+	/*
+	 * Check if the current request is coming from an allowed backend.
+	 *
+	 * The backends are sending the custom header "Spreed-Signaling-Random"
+	 * containing at least 32 bytes random data, and the header
+	 * "Spreed-Signaling-Checksum", which is the SHA256-HMAC of the random data
+	 * and the body of the request, calculated with the shared secret from the
+	 * configuration.
+	 *
+	 * @return bool
+	 */
+	private function validateBackendRequest($data) {
+		$random = $_SERVER['HTTP_SPREED_SIGNALING_RANDOM'];
+		if (empty($random) || strlen($random) < 32) {
+			return false;
+		}
+		$checksum = $_SERVER['HTTP_SPREED_SIGNALING_CHECKSUM'];
+		if (empty($checksum)) {
+			return false;
+		}
+		$hash = hash_hmac('sha256', $random . $data, $this->config->getSignalingSecret());
+		return hash_equals($hash, strtolower($checksum));
+	}
+
+	/**
+	 * Backend API to query information required for standalone signaling
+	 * servers.
+	 *
+	 * See sections "Backend validation" in
+	 * https://github.com/nextcloud/spreed/wiki/Spreed-Signaling-API
+	 *
+	 * @NoCSRFRequired
+	 * @PublicPage
+	 *
+	 * @param string $message
+	 * @return JSONResponse
+	 */
+	public function backend() {
+		$json = file_get_contents('php://input');
```

Comment:
can't you use `$this->request` ?

Reply:
`$this->request` is a `OCP\IRequest` which only provides access to the request variables / -headers but I need the body of the request. Maybe there is a better way to get that - if so I didn't find it ;-)

Reply:
Or you post a json body to the server using parameters instead of the body? ;)

Reply:
You mean sending a body `body=the-json-object` and then using the value of variable `body`? That sounds hackish ;-)